### PR TITLE
Rowwise Prune op (Add the test to OSS run_test), Make the op private.

### PR DESCRIPTION
**Pull request by radkris-git**
Summary: Refer to the title.

Test Plan: `buck test caffe2/test:pruning`

Differential Revision: D24230472

### CODE REVIEW

File: test/backward_compatibility/check_backward_compatibility.py [hunk 1]
```diff
@@ -111,6 +111,7 @@
     ("aten::_amp_non_finite_check_and_unscale_", datetime.date(9999, 1, 1)),
     ("aten::choose_qparams_optimized", datetime.date(2020, 10, 5)),
     ("aten::smooth_l1_loss_backward", datetime.date(2020, 10, 15)),
+    ("at::_rowwise_prune", datetime.date(9999, 1, 1)),
```
Comment (supriyar) [inline test/backward_compatibility/check_backward_compatibility.py:114]: This should be `rowwise_prune` instead

**Review by supriyar — CHANGES_REQUESTED** [review]
Need to update the backward compatibility test

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -1421,7 +1421,7 @@
     CPU: _embedding_bag_forward_only_cpu
     CUDA: _embedding_bag_forward_only_cuda
 
-- func: rowwise_prune(Tensor weight, Tensor mask, ScalarType compressed_indices_dtype) -> (Tensor, Tensor)
+- func: _rowwise_prune(Tensor weight, Tensor mask, ScalarType compressed_indices_dtype) -> (Tensor, Tensor)
```
Comment (gchanan) [inline aten/src/ATen/native/native_functions.yaml:1424]: I think it also makes sense to put this in the nn module, you can do that via "python_module: nn" -- note you'll have to access it via ._nn._C or something.

File: test/backward_compatibility/check_backward_compatibility.py (hunk 1 above)
Comment (radkris-git) [reply test/backward_compatibility/check_backward_compatibility.py]: Addressed

**Comment by radkris-git** [comment]
> Need to update the backward compatibility test

Addressed.

File: aten/src/ATen/native/native_functions.yaml (hunk 2 above)
Comment (radkris-git) [reply aten/src/ATen/native/native_functions.yaml:1494]: The plan is to silence this change as we don't want to expose this to PyTorch frontend as this operator doesn't have detailed documentation/API yet. Do you still think we need to add "python_module:nn" in this case?

Comment (raghuramank100) [reply aten/src/ATen/native/native_functions.yaml:1494]: @gchanan : If we are planning to access this for now only in C++, do we need to expose it via nn?

**Review by raghuramank100 — APPROVED** [review]

**Pull request closed without merging**